code/modules/mob/living/simple_animal/hostile/megafauna/onestarboss.dm

```diff
@@ -0,0 +1,178 @@
+/mob/living/simple_animal/hostile/megafauna/one_star
+	name = "Type - 0315"
+	desc = "Love and concrete."
+
+	faction = "hive"
+
+	icon = 'icons/mob/64x64.dmi'
+	icon_state = "onestar_boss_unpowered"
+	icon_living = "onestar_boss_unpowered"
+	icon_dead = "onestar_boss_wrecked"
+	pixel_x = -16
+	ranged = TRUE
+
+	health = 1700
+	maxHealth = 1700
+	break_stuff_probability = 95
+	stop_automated_movement = 1
+
+	melee_damage_lower = 10
+	melee_damage_upper = 20
+	megafauna_min_cooldown = 30
+	megafauna_max_cooldown = 60
+
+/mob/living/simple_animal/hostile/megafauna/one_star/death()
+	..()
+	walk(src, 0)
+
+/mob/living/simple_animal/hostile/megafauna/one_star/LoseTarget()
+	stance = HOSTILE_STANCE_IDLE
+	target_mob = null
+	icon_state = initial(icon_state)
+	walk(src, 0)
+
+/mob/living/simple_animal/hostile/megafauna/one_star/LostTarget()
+	stance = HOSTILE_STANCE_IDLE
+	icon_state = initial(icon_state)
+	walk(src, 0)
+
+/mob/living/simple_animal/hostile/megafauna/one_star/FindTarget()
+
+	var/atom/T = null
+	var/valueofview = 0
+
+	if(src.loc:luminosity > 0)
+		valueofview = 10
+	else
+		valueofview = 4
+
+	for(var/atom/A in ListTargets(valueofview))
+
+		if(A == src)
+			continue
+
+		var/atom/F = Found(A)
+		if(F)
+			T = F
+			break
+
+		if(isliving(A))
+			var/mob/living/L = A
+			if(L.faction == src.faction && !attack_same)
+				continue
+			else if(L in friends)
+				continue
+			else
+				if(!SA_attackable(L))
+					stance = HOSTILE_STANCE_ATTACK
+					T = L
+					break
+
+		else if(istype(A, /obj/mecha)) // Our line of sight stuff was already done in ListTargets().
+			var/obj/mecha/M = A
+			if (M.occupant)
+				stance = HOSTILE_STANCE_ATTACK
+				T = M
+				break
+
+		if(istype(A, /obj/machinery/bot))
+			var/obj/machinery/bot/B = A
+			if (B.health > 0)
+				stance = HOSTILE_STANCE_ATTACK
+				T = B
+				break
+	if(T != null)
+		icon_state = "onestar_boss"
+	else
+		icon_state = initial(icon_state)
+	return T
+
+
+/mob/living/simple_animal/hostile/megafauna/one_star/Life()
```

Comment:
This proc is completely same as parent hostile mob one. Why is it overriden at all?